File: Examples/CSharp/KnowledgeBase/Benchmarking/LoadLargeExcelScenario1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Cells.Examples.CSharp.KnowledgeBase.Benchmarking
{
    public class LoadLargeExcelScenario1
    {
        // ExStart:1
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);

            try
            {
                CreateAsposeCellsFile(dataDir + "Sample.xls", dataDir + "output_out.xls");
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
        }

        private static void CreateAsposeCellsFile(string filename_in, string filename_out)
        {
            DateTime start = DateTime.Now;
            string designerFile = filename_in;
            Workbook workbook = new Workbook(designerFile);
            for (int i = 0; i < 100; i++)
            {
                Worksheet ws = workbook.Worksheets[i];
                ws.Cells[0, 0].PutValue("Data" + i.ToString());
            }
            workbook.Save(filename_out);
            DateTime end = DateTime.Now;
            TimeSpan time = end - start;
            Console.WriteLine("File Updated! \n" + "Time consumed (Seconds): " + time.TotalSeconds.ToString());
        }
        // ExEnd:1
    }
}

File: Examples/VisualBasic/KnowledgeBase/Benchmarking/LoadLargeExcelScenario1.vb
Namespace KnowledgeBase.Benchmarking
    Public Class LoadLargeExcelScenario1
        ' ExStart:1
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            Try
                CreateAsposeCellsFile(dataDir & Convert.ToString("Sample.xls"), dataDir & Convert.ToString("output_out.xls"))
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try
        End Sub

        Private Shared Sub CreateAsposeCellsFile(filename_in As String, filename_out As String)
            Dim start As DateTime = DateTime.Now
            Dim designerFile As String = filename_in
            Dim workbook As New Workbook(designerFile)
            For i As Integer = 0 To 99
                Dim ws As Worksheet = workbook.Worksheets(i)
                ws.Cells(0, 0).PutValue("Data" + i.ToString())
            Next
            workbook.Save(filename_out)
            Dim [end] As DateTime = DateTime.Now
            Dim time As TimeSpan = [end] - start
            Console.WriteLine("File Updated! " & vbLf + "Time consumed (Seconds): " + time.TotalSeconds.ToString())
        End Sub
        ' ExEnd:1
    End Class
End Namespace